Number_types/include/CGAL/boost_mp.h

```diff
@@ -0,0 +1,537 @@
+// Copyright (c) 2017
+// INRIA Saclay-Ile de France (France),
+//
+// This file is part of CGAL (www.cgal.org); you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public License as
+// published by the Free Software Foundation; either version 3 of the License,
+// or (at your option) any later version.
+//
+// Licensees holding a valid commercial license may use this file in
+// accordance with the commercial license agreement provided with the software.
+//
+// This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
+// WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: LGPL-3.0+
+//
+// Author(s)     : Marc Glisse
+
+#ifndef CGAL_BOOST_MP_H
+#define CGAL_BOOST_MP_H
+
+#include <CGAL/config.h>
+// This could check BOOST_VERSION >= 105300, but before 1.56 there is no
+// implicit conversion from double, which makes it hard to use in CGAL.
+// It is easier to disable this number type completely for old versions.
+// Before 1.63, I/O is broken. If we have LEDA or GMP, we have a safer
+// alternative, but otherwise, without an alternative, we might as well still
+// use boost. Again, disabling the whole file is just the easy solution.
+#if !defined CGAL_DO_NOT_USE_BOOST_MP && BOOST_VERSION >= 105600 && (BOOST_VERSION >= 106300 || !defined CGAL_USE_GMP)
```

Comment:
I'd prefer to check for 1.70 only for MSVC, since other compilers don't seem to have the same issue. Boost 1.70 was hardened in a good way, but it is still unclear to me why MSVC tries to instantiate this while others don't.

Reply:
OK.

... We will discuss that again in another message of this PR (following https://github.com/CGAL/cgal/pull/3406#issuecomment-447284771).